docs/adr/0008-freeing-objects-and-resources.md

```diff
@@ -0,0 +1,146 @@
+# Freeing objects and resources
+
+* Status: Accepted
+* Deciders: Uniffi developers, as approved in PR-1787
+* Date: 2023-10-18
+
+Prevous discussion: [PR 1394](https://github.com/mozilla/uniffi-rs/issues/1394), [Issue 8](https://github.com/mozilla/uniffi-rs/issues/8)
+
+## Context and Problem Statement
+
+## Releasing References
+When a Rust object is  passed to the foreign side, the foreign code stores reference to a Rust `Arc<>`.
+Conversely, foreign objects passed into Rust (eg, callbacks or trait implementations) cause Rust to hold a reference to the foreign object.
+
+UniFFI needs to ensure that when the generated objects are destroyed, the reference on the other side of the FFI is released.
```

Comment:
Form nit: Is this the problem statement? 

I had to read the Context and Problem statement at least twice to identify this sentence as what the ADR considers the nub of the problem. I'm still unsure.